dns/name.py

```diff
@@ -21,88 +21,119 @@
 @type empty: dns.name.Name object
 """
 
-import cStringIO
+from io import BytesIO
 import struct
 import sys
 import copy
-
-if sys.hexversion >= 0x02030000:
-    import encodings.idna
+import encodings.idna
 
 import dns.exception
 import dns.wiredata
 
+from ._compat import long, binary_type, text_type
+
+
+try:
+    maxint = sys.maxint
+except:
+    maxint = (1 << (8 * struct.calcsize("P"))) / 2 - 1
+
 NAMERELN_NONE = 0
 NAMERELN_SUPERDOMAIN = 1
 NAMERELN_SUBDOMAIN = 2
 NAMERELN_EQUAL = 3
 NAMERELN_COMMONANCESTOR = 4
 
+
 class EmptyLabel(dns.exception.SyntaxError):
+
     """A DNS label is empty."""
 
+
 class BadEscape(dns.exception.SyntaxError):
+
     """An escaped code in a text format of DNS name is invalid."""
 
+
 class BadPointer(dns.exception.FormError):
+
     """A DNS compression pointer points forward instead of backward."""
 
+
 class BadLabelType(dns.exception.FormError):
+
     """The label type in DNS name wire format is unknown."""
 
+
 class NeedAbsoluteNameOrOrigin(dns.exception.DNSException):
+
     """An attempt was made to convert a non-absolute name to
     wire when there was also a non-absolute (or missing) origin."""
 
+
 class NameTooLong(dns.exception.FormError):
+
     """A DNS name is > 255 octets long."""
 
+
 class LabelTooLong(dns.exception.SyntaxError):
+
     """A DNS label is > 63 octets long."""
 
+
 class AbsoluteConcatenation(dns.exception.DNSException):
+
     """An attempt was made to append anything other than the
     empty name to an absolute DNS name."""
 
+
 class NoParent(dns.exception.DNSException):
+
     """An attempt was made to get the parent of the root name
     or the empty name."""
 
-_escaped = {
-    '"' : True,
-    '(' : True,
-    ')' : True,
-    '.' : True,
-    ';' : True,
-    '\\' : True,
-    '@' : True,
-    '$' : True
-    }
+_escaped = [c for c in bytearray(b'"().;\\@$')]
```

Comment:
doesn't it work also if you just have this:

_escaped = bytearray(b'"().;\\@$')